packages/store/src/storage.ts

```diff
@@ -1905,6 +1915,25 @@ class Storage {
       return cacheManifest;
     }
   }
+
+  private async getTarballStats(version: Version, buffer: Buffer): Promise<TarballDetails> {
+    if (
+      version.dist == undefined ||
+      version.dist?.fileCount == undefined ||
+      version.dist?.unpackedSize == undefined
+    ) {
+      debug('tarball stats not found, calculating');
+      try {
+        return await getTarballDetails(buffer);
+      } catch (err: any) {
+        logger.error({ err: err.message }, 'getting tarball details has failed: @{err}');
+        throw err;
```

Comment:
🤔 not very sure about throwing here might have side effects or just loggin the error

Reply:
I moved the getTarballStats call into an existing try catch. That should be better